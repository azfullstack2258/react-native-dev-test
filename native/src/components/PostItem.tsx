import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import moment from 'moment';

interface IPostListItemProps {
  id: string;
  title: string;
  publishedAt: Date;
  author: string;
  onPress: (id: string, title: string) => void;
}

const PostListItem: React.FC<IPostListItemProps> = ({
  id,
  title,
  publishedAt,
  author,
  onPress,
}) => {
  const handlePress = () => {
    onPress(id, title);
  };

  return (
    <TouchableOpacity
      key={id}
      style={styles.itemContainer}
      onPress={handlePress}
    >
      <Text style={styles.postTitle}>{title}</Text>
      <Text style={styles.postDetail}>
        {`${author}, ${moment(publishedAt).format('YYYY/MM/DD')}`}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  itemContainer: {
    paddingVertical: 10,
    paddingHorizontal: 20,
    borderBottomColor: '#e0e0e0',
    borderBottomWidth: 1,
    borderStyle: 'solid',
  },
  postTitle: {
    fontSize: 20,
    fontWeight: '500',
  },
  postDetail: {
    fontStyle: 'italic',
    fontSize: 10,
  },
});

export default PostListItem;
