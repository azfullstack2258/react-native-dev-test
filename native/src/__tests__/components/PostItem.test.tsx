import React from 'react';
import renderer from 'react-test-renderer';
import PostItem from '../../components/PostItem';

const mockPostData = {
  id: 'a6cc3a8c-db06-4314-8a67-64687f065fe6',
  title: 'CQB',
  publishedAt: new Date(),
  author: 'Jeff Woolnough',
};

describe('Post Item', () => {
  it('can render post item', () => {
    const postItem = renderer
      .create(<PostItem {...mockPostData} onPress={() => {}} />)
      .toJSON();
    expect(postItem).toMatchSnapshot();
  });
});
