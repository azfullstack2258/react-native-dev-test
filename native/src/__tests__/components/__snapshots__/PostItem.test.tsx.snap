// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Item can render post item 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#e0e0e0",
      "borderBottomWidth": 1,
      "borderStyle": "solid",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "500",
      }
    }
  >
    CQB
  </Text>
  <Text
    style={
      Object {
        "fontSize": 10,
        "fontStyle": "italic",
      }
    }
  >
    Jeff Woolnough, 2020/04/26
  </Text>
</View>
`;
